src/telemetry/Manager.cc

```diff
@@ -38,6 +69,415 @@ void Manager::InitPostBrokerSetup(broker::endpoint& ep)
+	r->Assign(help_text_idx, make_intrusive<zeek::StringVal>(broker::telemetry::helptext(family)));
+	r->Assign(unit_idx, make_intrusive<zeek::StringVal>(broker::telemetry::unit(family)));
+	r->Assign(is_total_idx, val_mgr->Bool(broker::telemetry::is_sum(family)));
+
+	auto label_names_vec = make_intrusive<zeek::VectorVal>(string_vec_type);
+	for ( const auto& l : broker::telemetry::label_names(family) )
+		label_names_vec->Append(make_intrusive<StringVal>(l));
+	r->Assign(labels_idx, label_names_vec);
+
+	// This is mapping Manager.h enums to bif values depending on
+	// the template type and whether this is a counter, gauge or
+	// histogram.
+	zeek_int_t metric_type_int = -1;
+	if constexpr ( std::is_same<T, double>::value )
+		{
+		if ( metric_type == MetricType::Counter )
+			metric_type_int = BifEnum::Telemetry::MetricType::DOUBLE_COUNTER;
+		else if ( metric_type == MetricType::Gauge )
+			metric_type_int = BifEnum::Telemetry::MetricType::DOUBLE_GAUGE;
+		else if ( metric_type == MetricType::Histogram )
+			metric_type_int = BifEnum::Telemetry::MetricType::DOUBLE_HISTOGRAM;
+		}
+	else
+		{
+		if ( metric_type == MetricType::Counter )
+			metric_type_int = BifEnum::Telemetry::MetricType::INT_COUNTER;
+		else if ( metric_type == MetricType::Gauge )
+			metric_type_int = BifEnum::Telemetry::MetricType::INT_GAUGE;
+		else if ( metric_type == MetricType::Histogram )
+			metric_type_int = BifEnum::Telemetry::MetricType::INT_HISTOGRAM;
+		}
+
+	if ( metric_type_int < 0 )
+		reporter->FatalError("Unable to lookup metric type %d", int(metric_type));
+
+	r->Assign(metric_type_idx,
+	          zeek::BifType::Enum::Telemetry::MetricType->GetEnumVal(metric_type_int));
+
+	metric_opts_cache.insert({family, r});
+
+	return r;
+	}
+
+zeek::RecordValPtr Manager::CollectedValueMetric::AsMetricRecord() const
+	{
+	static auto string_vec_type = zeek::id::find_type<zeek::VectorType>("string_vec");
+	static auto metric_record_type = zeek::id::find_type<zeek::RecordType>("Telemetry::Metric");
+	static auto opts_idx = metric_record_type->FieldOffset("opts");
+	static auto labels_idx = metric_record_type->FieldOffset("labels");
+	static auto value_idx = metric_record_type->FieldOffset("value");
+	static auto count_value_idx = metric_record_type->FieldOffset("count_value");
+
+	auto r = make_intrusive<zeek::RecordVal>(metric_record_type);
+
+	auto label_values_vec = make_intrusive<zeek::VectorVal>(string_vec_type);
+	for ( const auto& l : label_values )
+		label_values_vec->Append(make_intrusive<StringVal>(l));
+	r->Assign(labels_idx, label_values_vec);
+
+	auto fn = [&](auto val)
+	{
+		using val_t = decltype(val);
+		auto opts_record = telemetry_mgr->GetMetricOptsRecord<val_t>(metric_type, family);
+		r->Assign(opts_idx, opts_record);
+		r->Assign(value_idx, as_double_val(val));
+		if constexpr ( std::is_same_v<val_t, int64_t> )
+			r->Assign(count_value_idx, val_mgr->Count(val));
+	};
```

Comment:
```suggestion
	};
	
```

Reply:
There was one more that matched the extra line after lamda. That's fixed, too.